### PR TITLE
Update the UAA-based login flow to discover org id

### PR DESCRIPTION
**Pull request by vuil**
best case effort to retrieve org name/id from the metadata file from a predetermined URL.

Env vars to support setting of these values are still supported as a fallback.

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

1. Updated and ran unit tests.

2. Verify that tanzu login to a uaa-based endpoint retrieves the org id:

```

> tanzu login --endpoint https://some.tpsm.host --insecure-skip-tls-verify
[i] Opening the browser window to complete the login
Log in by visiting this link:

    https://some.tpsm.host/auth/oauth/authorize?client_id=tp_cli_app&code_challenge=hKNsxxxx&code_challenge_method=S256&redirect_uri=http%3A%2F%2F127.0.0.1%3A62785%2Fcallback&response_type=code&state=c5f12xxxxx

    Optionally, paste your authorization code: [...]


[ok] Successfully logged in to 'https://some.tpsm.host' and created a tanzu context
> tanzu context current
  Name:            tpsm-4b01bf5c
  Type:            tanzu
  Organization:    Tanzu Platform Self Managed (703dde2e-12df-41c4-af07-153989894d1d)
  Project:         none set
  Kube Config:     /Users/foo/.config/tanzu/kube/config
  Kube Context:    tanzu-cli-tpsm-4b01bf5c
tanzu-cli (discover-sm-org)> grep 703dde2e-12df-41c4-af07-153989894d1d /Users/foo/.config/tanzu/kube/config
    server: https://some.tpsm.host/org/703dde2e-12df-41c4-af07-153989894d1d

```

### Release note

```release-note
Streamline UAA-based Tanzu Platform login to not require Org ID
```

### CODE REVIEW

**Review by prkalle — APPROVED** [review]
LGTM. Thanks!

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks

File: pkg/command/context.go [hunk 1]
```diff
@@ -649,31 +649,98 @@ func globalTanzuLogin(c *configtypes.Context, generateContextNameFunc func(orgNa
...
 		orgName = uaaOrgNameValue
-	} else if orgID != "" {
-		orgName = orgID
 	}
 
+	return orgID, orgName
+}
+
+func globalTanzuLoginUAA(c *configtypes.Context, generateContextNameFunc func(orgName, endpoint string, isStaging bool) string) error {
+	uaaEndpoint := c.AdditionalMetadata[config.TanzuAuthEndpointKey].(string)
```
Comment (anujc25) [inline pkg/command/context.go:733]: Any possibility that `c.AdditionalMetadata[config.TanzuAuthEndpointKey]` is not configured and this causes a panic?

**Review by prkalle — APPROVED** [review]
LGTM.

**Review by anujc25 — APPROVED** [review]
LGTM. Thanks

File: pkg/command/context.go [hunk 2]
```diff
@@ -649,31 +649,98 @@ func globalTanzuLogin(c *configtypes.Context, generateContextNameFunc func(orgNa
...
 
-	claims, err := doInteractiveLoginAndUpdateContext(c, uaaEndpoint)
+type DeploymentMetadataOrganization struct {
+	DisplayName string `json:"displayName"`
+	ID          string `json:"id"`
+	Name        string `json:"name"`
+}
+
+func getOrgFromSelfManagedEndpoint(c *configtypes.Context) (string, string, error) {
+	hubEndpoint := c.AdditionalMetadata[config.TanzuHubEndpointKey].(string)
```
Comment (anujc25) [inline pkg/command/context.go:664]: Any possibility that `c.AdditionalMetadata[config.TanzuHubEndpointKey]` is not configured and this causes a panic?

Comment (vuil) [reply pkg/command/context.go]: good point. To be safe, updated to check for existence as well.

**Review by anujc25 — APPROVED** [review]

**Pull request merged**